src/dmd/parse.d

```diff
@@ -242,7 +279,7 @@ final class Parser(AST) : Lexer
         //printf("Parser::Parser()\n");
         scanloc = loc;
 
-        if (loc.filename)
+        if (!writeMixin(input, scanloc) && loc.filename)
```

Comment:
Not too happy about `writeMixin` being in the parse module. Maybe it would be better to do this (and the existing pseudo-filename creation) at the callsite with a small helper function.

Reply:
I tried that originally, but that constructor is called from `MixinStatement` and `MixinDeclaraion` and I didn't really want to duplicate the code.